subprojects/docs/src/docs/userguide/multi_project_builds.adoc

```diff
@@ -48,403 +48,660 @@ Configuration on demand mode attempts to configure only projects that are releva
 * A task requested via task path from the command line (or Tooling API) causes the relevant project to be configured. For example, building 'projectA:projectB:someTask' causes configuration of projectB.
 
 
-Eager to try out this new feature? To configure on demand with every build run see <<sec:gradle_configuration_properties>>. To configure on demand just for a given build, see <<sec:command_line_performance>>.
+Eager to try out this new feature? To configure on demand with every build run see <<build_environment.adoc#sec:gradle_configuration_properties,Gradle properties>>. To configure on demand just for a given build, see <<command_line_interface.adoc#sec:command_line_performance,command-line performance-oriented options>>.
 
 [[sec:defining_common_behavior]]
-==== Defining common behavior
+=== Defining common behavior
 
 Let's look at some examples with the following project tree. This is a multi-project build with a root project named `water` and a subproject named `bluewhale`.
 
-++++
-<sample id="multiprojectFirstExample" dir="userguide/multiproject/firstExample/water" includeLocation="true" title="Multi-project tree - water &amp; bluewhale projects">
-                <layout>
-                    build.gradle
-                    settings.gradle
-                    bluewhale/
-                </layout>
-                <sourcefile file="settings.gradle"/>
-            </sample>
-++++
+=== Example: Multi-project tree - water &amp; bluewhale projects
```

Comment:
:x: Missing note admonition about where to find the sample in the `-all` distribution:

![image](https://user-images.githubusercontent.com/132773/42990901-5fcfeb7e-8c04-11e8-9d23-f631983ced58.png)